src/components/NewCommentForm.tsx

```
@@ -1,8 +1,94 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { CommentData } from '../types/Comment';
+import classNames from 'classnames';
+
+type Props = {
+  onSubmit: (data: CommentData) => Promise<void>;
+  isSubmitting: boolean;
+};
+
+type Errors = Partial<Record<keyof CommentData, string>>;
+
+export const NewCommentForm: React.FC<Props> = ({ onSubmit, isSubmitting }) => {
+  const [formData, setFormData] = useState<CommentData>({
+    name: '',
+    email: '',
+    body: '',
+  });
+
+  const [errors, setErrors] = useState<Errors>({});
+  const [wasSubmitted, setWasSubmitted] = useState(false);
+
+  const validate = (data: CommentData): Errors => {
+    const newErrors: Errors = {};
+
+    if (!data.name.trim()) {
+      newErrors.name = 'Name is required';
+    }
+
+    if (!data.email.trim()) {
+      newErrors.email = 'Email is required';
+    }
+
+    if (!data.body.trim()) {
+      newErrors.body = 'Enter some text';
+    }
+
+    return newErrors;
+  };
+
+  useEffect(() => {
+    if (wasSubmitted) {
+      setErrors(validate(formData));
+    }
+  }, [formData, wasSubmitted]);
+
+  const handleChange = (
+    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+  ) => {
+    const { name, value } = event.target;
+
+    setFormData(current => ({
+      ...current,
+      [name]: value,
+    }));
+  };
+
+  const handleSubmit = async (event: React.FormEvent) => {
+    event.preventDefault();
+    setWasSubmitted(true);
+
+    const validationErrors = validate(formData);
+
+    setErrors(validationErrors);
+
+    const hasErrors = Object.values(validationErrors).some(Boolean);
+
+    if (hasErrors) {
+      return;
+    }
+
+    try {
+      await onSubmit(formData);
+
+      setFormData(prev => ({
+        ...prev,
+        body: '',
+      }));
+
+      setErrors({});
+      setWasSubmitted(false);
+    } catch {}
```

This function still contains an empty catch block which swallows any error thrown by `onSubmit`. That breaks the requirement: "(*) Handle `Add` and `Delete` errors so the user can retry." The previous review explicitly asked to remove the empty try/catch so errors can propagate to the parent (or handle the error properly here and surface it). Either remove the empty `catch {}` so the error reaches the parent, or handle the error (set an error state and allow retry) and rethrow if needed. Swallowing errors prevents the app from showing an error notification and retrying.